### PR TITLE
Add a helper function get_executor for retrieving executors from NEMORUN_HOME

### PR DESCRIPTION
**Pull request by hemildesai**
Similar to workspace.py, but specifically for executors. 

Example `executors.py`
```python
import nemo_run as run

EXECUTOR_MAP = {
    "local": run.LocalExecutor()
}
```

Usage:
```python
run.get_executor("local")
```

### CODE REVIEW

**Comment by marcromeyn** [comment]
What's the advantage over workspace.py? Could we combine the 2 somehow?

**Comment by hemildesai** [comment]
> What's the advantage over workspace.py? Could we combine the 2 somehow?

This is specifically for executors. It can be used with/without CLI as per the user requirements. Maybe we can separate workspace.py into `factories.py` and `executors.py` to make it consistent.

**Comment by marcromeyn** [comment]
Why not just have `workspace.py`? We changed the API for that so right now it's just a python file that gets imported so I don't see a reason why we would want to split it up in multiple files. A user can always do that themselves by importing other things inside the workspace.py

**Comment by hemildesai** [comment]
> Why not just have `workspace.py`? We changed the API for that so right now it's just a python file that gets imported so I don't see a reason why we would want to split it up in multiple files. A user can always do that themselves by importing other things inside the workspace.py

Since we will try to split up config and execution, it makes sense to have executors.py and factories.py seperate since those files will be loaded in separate places. For eg, for loading executors, you shouldn't have to load the config factories. This will also lead to faster load times for the specific use case.

File: src/nemo_run/core/execution/base.py [hunk 1]
```diff
@@ -226,3 +227,49 @@ def package_configs(self, *cfgs: tuple[str, str]) -> list[str]:
         return filenames
 
     def cleanup(self, handle: str): ...
+
+
+def get_executor(
```
Comment (marcromeyn) [inline src/nemo_run/core/execution/base.py:232]: Should we rename this to `import_executor`?

Comment (hemildesai) [reply src/nemo_run/core/execution/base.py]: Done in https://github.com/NVIDIA/NeMo-Run/pull/105/commits/b3146e64f12f11c0a9e4400efae8d0618f93eb23

**Review by marcromeyn — APPROVED** [review]

**Pull request merged**